lib/src/dartdoc_options.dart

```diff
@@ -9,17 +9,707 @@
+        returnData = <String>[];
+        for (var item in yamlData as YamlList) {
+          returnData.add(item.toString());
+        }
+      }
+    } else if (_isMapString) {
+      if (yamlData is YamlMap) {
+        returnData = <String, String>{};
+        for (MapEntry entry in yamlData.entries) {
+          returnData[entry.key.toString()] = entry.value.toString();
+        }
+      }
+    } else if (_isDouble) {
+      if (yamlData is num) {
+        returnData = (yamlData as num).toDouble();
+      }
+    } else if (_isInt || _isString || _isBool) {
+      if (yamlData is T) {
+        returnData = yamlData;
+      }
+    } else {
+      throw new UnsupportedError("Type ${T} is not supported");
+    }
+    return new _OptionValueWithContext(returnData as T, contextPath,
+        definingFile: 'dartdoc_options.yaml');
+  }
+
+  _YamlFileData _yamlAtDirectory(Directory dir) {
+    List<String> canonicalPaths = [pathLib.canonicalize(dir.path)];
+    if (!_yamlAtCanonicalPathCache.containsKey(canonicalPaths.first)) {
+      _YamlFileData yamlData = new _YamlFileData(
+          new Map(), pathLib.canonicalize(Directory.current.path));
+      if (dir.existsSync()) {
+        File dartdocOptionsFile;
+
+        while (true) {
+          dartdocOptionsFile =
+              new File(pathLib.join(dir.path, 'dartdoc_options.yaml'));
+          if (dartdocOptionsFile.existsSync() ||
+              pathLib.equals(dir.parent.path, dir.path)) break;
+          dir = dir.parent;
+          canonicalPaths.add(pathLib.canonicalize(dir.path));
+        }
+        if (dartdocOptionsFile.existsSync()) {
+          yamlData = new _YamlFileData(
+              loadYaml(dartdocOptionsFile.readAsStringSync()),
+              pathLib.canonicalize(dir.path));
+        }
+      }
+      canonicalPaths.forEach((p) => _yamlAtCanonicalPathCache[p] = yamlData);
+    }
+    return _yamlAtCanonicalPathCache[canonicalPaths.first];
+  }
+}
+
+/// Mixin class implementing command-line arguments for [DartdocOption].
+abstract class _DartdocArgOption<T> implements DartdocOption<T> {
+  /// For [ArgsParser], set to true if the argument can be negated with --no on the command line.
+  bool get negatable;
+
+  /// For [ArgsParser], set to true if a single string argument will be broken into a list on commas.
+  bool get splitCommas;
+
+  /// For [ArgsParser], set to true to hide this from the help menu.
+  bool get hide;
+
+  /// For [ArgsParser], set to a single character to have a short version of the command line argument.
+  String get abbr;
+
+  /// valueAt for arguments ignores the [dir] parameter and only uses command
+  /// line arguments and the current working directory to resolve the result.
+  @override
+  T valueAt(Directory dir) => _valueAtFromArgs() ?? defaultsTo;
+
+  /// For passing in to [int.parse] and [double.parse] `onError'.
+  _throwErrorForTypes(String value) {
+    String example;
+    if (defaultsTo is Map) {
+      example = "key::value";
```

Comment:
I think you're preferring single quote strings elsewhere...

Reply:
done